src/Symfony/Component/HttpKernel/Tests/EventListener/TestSessionListenerTest.php

```diff
@@ -106,6 +106,36 @@ public function testEmptySessionWithNewSessionIdDoesSendCookie()
         $this->assertNotEmpty($response->headers->getCookies());
     }
 
+    /**
+     * @dataProvider anotherCookieProvider
+     */
+    public function testSessionWithNewSessionIdAndNewCookieDoesNotSendAnotherCookie($existing, array $expected)
+    {
+        $this->sessionHasBeenStarted();
+        $this->sessionIsEmpty();
+        $this->fixSessionId('456');
+
+        $kernel = $this->getMockBuilder('Symfony\Component\HttpKernel\HttpKernelInterface')->getMock();
+        $request = Request::create('/', 'GET', array(), array('MOCKSESSID' => '123'));
+        $event = new GetResponseEvent($kernel, $request, HttpKernelInterface::MASTER_REQUEST);
+        $this->listener->onKernelRequest($event);
+
+        $response = new Response('', 200, array('Set-Cookie' => $existing));
+
+        $response = $this->filterResponse(new Request(), HttpKernelInterface::MASTER_REQUEST, $response);
+
+        $this->assertSame($expected, $response->headers->get('Set-Cookie', null, false));
```

Comment:
It's not really clear to me what this PR for. You state that it is to make it compatible with the `CookieClearingLogoutHandler`, but then I would expect a test that shows that no cookies are sent. However, you've added no tests that show this.

Furthermore, I doubt if this fix is the correct way to do it. It feels like it's battling symptoms instead of the root problem, but I can't quite lay my finger on it.

Reply:
I still expect the original cookie to be sent (ie the one set by `CookieClearingLogoutHandler`).

Reply:
But the `CookieClearingLogoutHandler` doesn't set cookies, it _removes_ cookies. That's the whole point.

Reply:
To 'remove' a cookie you have to set a new expired one, hence: https://github.com/symfony/symfony/blob/6064cfe0162602701bff943ea5e21481d264906b/src/Symfony/Component/HttpFoundation/ResponseHeaderBag.php#L248-L251

Reply:
aaah, check, I get it now.

In that case, LGTM